app/src/main/res/values/strings.xml

```diff
@@ -261,4 +261,52 @@
   <string name="notes_deletion_none_found">No notes found for deletion</string>
   <string name="notes_deletion_successful">Entire notes folder deleted</string>
   <string name="notes_deletion_unsuccessful">Some files not deleted</string>
+  <string name="discovery_initiated">Discovery Initiated</string>
+  <string name="discovery_failed">Discovery Failed: %s</string>
+  <string name="wifi_p2p_internal_error">Internal error</string>
+  <string name="wifi_p2p_framework_busy">Framework busy, unable to service request</string>
+  <string name="wifi_p2p_unsupported">P2P unsupported on this device</string>
+  <string name="wifi_p2p_unknown_error">Unknown error code - %i</string>
+  <string name="channel_lost">Channel lost, trying again</string>
+  <string name="severe_loss_error">Severe! Try Disable/Re-enable WiFi P2P</string>
+  <string name="abort_connection">Aborting connection</string>
+  <string name="abort_failed">Connect abort request failed. Reason: %s</string>
+  <string name="connection_failed">Connection failed: %s</string>
+  <string name="permission_rationale_location">Location permission is required by Android to allow the app to detect peer devices</string>
+  <string name="permission_rationale_storage">Storage permissions required for accessing and storing ZIM files</string>
```

Comment:
We should already have storage permissions?

Reply:
So starting from Marshmallow if I'm not wrong the user can choose to disable some permissions, so we need to programmatically check that they've been granted before using them (& request them if not granted). 
Now whenever I tried to change permissions while the app is running led to the app getting restarted on my phone (which led to granting of storage permission for "accessing offline content", and hence not requiring my code to request). But this behaviour is not guaranteed (I also came across complaints on SO), so I need a contingency to ask for permissions.

I didn't use the existing `To access offline content we need access to your storage` string because it didn't seem as explanatory. Should I use it instead?

Reply:
Yeah, I would use it to cut back on strings.

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/de956fc663be0b4e6af31d997aca1547b5c826fe)